tests/mail/tests.py

```diff
@@ -1552,6 +1552,27 @@ def test_all_params_can_be_set_before_send(self):
         self.assertIs(email.get_connection(), new_connection)
         self.assertNotIn("original", message.as_string())
 
+    def test_unicode_attachment_encoding(self):
+        """
+        Test that attaching a message containing non-ASCII characters does not
+        raise a UnicodeEncodeError.
+        """
+        att = """\
+        Subject: attachment
+        Content-Type: text/plain; charset=utf-8
+        Content-Transfer-Encoding: 8bit
+
+        ¡8-bit content!
+        """.encode()
```

Comment:
Technically, a message's headers can't have spaces before them. dedent() is an easy way to fix that. (Also, the Subject header isn't really relevant to the bug.)

```suggestion
        att = dedent(
            """\
            Content-Type: text/plain; charset=utf-8
            Content-Transfer-Encoding: 8bit
    
            ¡8-bit content!
            """
        ).encode()
```